src/main/java/sirius/biz/translations/BasicTranslations.java

```diff
@@ -0,0 +1,223 @@
+     */
+    @AfterDelete
+    protected abstract void removeTranslations();
+
+    /**
+     * Provides a validator so translations can only be added for supported languages.
+     *
+     * @param lang the language code in question
+     * @return true if language is supported by the system, false otherwise
+     */
+    protected boolean isSupportedLanguage(String lang) {
+        return supportedLanguages.contains(lang);
+    }
+
+    /**
+     * Updates the translation for the given field and language, or adds a new translation if it does not exist.
+     * <p>
+     * Note that an existing translation is deleted when calling this with empty text.
+     * <p>
+     * Also note that this will emit an {@link sirius.biz.protocol.JournalEntry} to
+     * {@link sirius.biz.protocol.Journaled} owners keeping track of the changes made to their translations.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  language code
+     * @param text  translated text for the given language
+     */
+    public void updateText(@Nonnull Mapping field, String lang, String text) {
+        if (Strings.isEmpty(text)) {
+            deleteText(field, lang);
+            return;
+        }
+
+        T translation = findOrCreateTranslation(field, lang, text);
+        translation.getTranslationData().setText(text);
+        updateTranslation(translation);
+
+        if (owner instanceof Journaled) {
+            JournalData.addJournalEntry(owner,
+                                        String.format("Updated translated text for %s (%s): '%s'",
+                                                      field.getName(),
+                                                      lang,
+                                                      text));
+        }
+    }
+
+    /**
+     * Forwards updates on the given translation entity to the database in use.
+     *
+     * @param translation the translation entity to update
+     */
+    protected abstract void updateTranslation(T translation);
+
+    /**
+     * Deletes the translation for the given field and language.
+     * <p>
+     * Note that this will also emit an {@link sirius.biz.protocol.JournalEntry} to
+     * {@link sirius.biz.protocol.Journaled} owners keeping track of the changes made to their translations.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  code of the language to be deleted
+     */
+    public abstract void deleteText(@Nonnull Mapping field, String lang);
+
+    /**
+     * Deletes all translations for the given field.
+     * <p>
+     * Note that this will also emit an {@link sirius.biz.protocol.JournalEntry} to
+     * {@link sirius.biz.protocol.Journaled} owners keeping track of the changes made to their translations.
+     *
+     * @param field {@link Mapping} of the translated field
+     */
+    public abstract void deleteAllTexts(@Nonnull Mapping field);
```

Comment:
shouldn't we also place a @AfterDelete handler here, which wipes all texts (without logging to the journal)?

Reply:
the @AfterDelete is handled by removeTranslations()

Reply:
nvm - totally didn't see that...